From 6600a270c468147dc6f445698053f8137c6a1a9c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sven=20K=C3=B6hler?= <sven.koehler@gmail.com>
Date: Mon, 7 Aug 2017 18:49:20 +0200
Subject: [PATCH 186/215] Fix dependencies broken since driver was renamed

---
 drivers/mmc/host/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/host/Kconfig b/drivers/mmc/host/Kconfig
index e577d20..324da89 100644
--- a/drivers/mmc/host/Kconfig
+++ b/drivers/mmc/host/Kconfig
@@ -25,7 +25,7 @@ config MMC_BCM2835_DMA
 
 config MMC_BCM2835_PIO_DMA_BARRIER
 	int "Block count limit for PIO transfers"
-	depends on MMC_BCM2835 && MMC_BCM2835_DMA
+	depends on MMC_BCM2835_MMC && MMC_BCM2835_DMA
 	range 0 256
 	default 2
 	help
-- 
1.9.1

