From cddcaae4cce990d4ac8d63eea2f47dcc45366bda Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Mon, 11 Jun 2018 17:29:55 +0100
Subject: [PATCH 145/220] configfs: hack: make it build

---
 drivers/of/configfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/of/configfs.c b/drivers/of/configfs.c
index 178f062..9f608b2 100644
--- a/drivers/of/configfs.c
+++ b/drivers/of/configfs.c
@@ -64,7 +64,7 @@ static int create_overlay(struct cfs_overlay_item *overlay, void *blob)
 	}
 	pr_debug("%s: resolved OK\n", __func__);
 
-	err = of_overlay_apply(overlay->overlay, &overlay->ov_id);
+	err = -1; // TODO: fix: of_overlay_apply(overlay->overlay, &overlay->ov_id);
 	if (err < 0) {
 		pr_err("%s: Failed to create overlay (err=%d)\n",
 				__func__, err);
-- 
1.9.1

