From f083227bc3975e3efa36655834c750e2ee42a8e7 Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Sun, 17 Jun 2018 13:22:41 +0100
Subject: [PATCH 162/220] ASoC: adau1977-adc: change codec to component

Signed-off-by: popcornmix <popcornmix@gmail.com>
---
 sound/soc/bcm/adau1977-adc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/bcm/adau1977-adc.c b/sound/soc/bcm/adau1977-adc.c
index f3d7e5d..83225b9 100644
--- a/sound/soc/bcm/adau1977-adc.c
+++ b/sound/soc/bcm/adau1977-adc.c
@@ -43,7 +43,7 @@ static int eval_adau1977_init(struct snd_soc_pcm_runtime *rtd)
 	if (ret < 0)
 		return ret;
 
-	return snd_soc_codec_set_sysclk(rtd->codec, ADAU1977_SYSCLK,
+	return snd_soc_component_set_sysclk(codec_dai->component, ADAU1977_SYSCLK,
 			ADAU1977_SYSCLK_SRC_MCLK, 11289600, SND_SOC_CLOCK_IN);
 }
  
-- 
1.9.1

